lib/class-wp-rest-term-search-handler.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * REST API: WP_REST_Terms_Search_Handler class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 5.5.0
+ */
+
+/**
+ * Core class representing a search handler for term in the REST API.
+ *
+ * @since 5.5.0
+ *
+ * @see WP_REST_Search_Handler
+ */
+class WP_REST_Term_Search_Handler extends WP_REST_Search_Handler {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 5.0.0
+	 */
+	public function __construct() {
+		$this->type = 'term';
+
+		// Support all public post types except attachments.
+		$this->subtypes = array_values(
+			get_taxonomies(
+				array(
+					'public' => true,
+				),
+				'names'
+			)
+		);
+	}
+
+	/**
+	 * Searches the object type content for a given search request.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array Associative array containing an `WP_REST_Search_Handler::RESULT_IDS` containing
+	 *               an array of found IDs and `WP_REST_Search_Handler::RESULT_TOTAL` containing the
+	 *               total count for the matching search results.
+	 */
+	public function search_items( WP_REST_Request $request ) {
+		$term_search = '';
+		if ( ! empty( $request['search'] ) ) {
+			$term_search = $request['search'];
+		}
+
+		$term_search = apply_filters( 'rest_term_search_query', $term_search, $request );
```

Comment:
So this filter should be filtering the list of args passed to `get_terms`, not just the search term.

Reply:
Done in 07dd970e92.